src/thruster.cpp

```diff
@@ -180,7 +180,8 @@ void Thruster::Update()
     // TODO: (If needed): Optimize this section so that the force is only
     // calculated once per thruster per message instead of once per frame per
     // thruster.
-    if(last_thruster_msg.data.size() == 8)
+    if(last_thruster_msg.data.size() == 8 &&
+            ros::Time::now() - last_msg_receive_time < ros::Duration(0.5))
```

Comment:
Maybe make the timeout duration a named constant or a parameter? That way a new Duration object isn't being created every time this line gets run.